### PR TITLE
Unique create wallet button

### PR DESCRIPTION
**Pull request by brunobar79**
This PR makes a few changes in order to simplify wallet creation.

- There's only one button to create wallet
- New wallets will be created in what we call the "primary wallet"
- The primary wallet is the one that's created by the app on start, or the first one with a seed phrase that the user imported
- Added migration to make this change backwards compatible

### CODE REVIEW

File: src/model/migrations.js [hunk 1]
```diff
@@ -15,7 +15,11 @@ export default async function runMigrations() {
   const currentVersion = Number(await getMigrationVersion());
   const migrations = [];
 
-  /***** Migration v0 starts here  *****/
+  /*
+   *************** Migration v1 ******************
```
Comment (brunobar79) [inline src/model/migrations.js:19]: it should say v0 here

File: src/model/migrations.js [hunk 2]
```diff
@@ -51,20 +58,76 @@ export default async function runMigrations() {
...
+  /*
+   *************** Migration v2 ******************
+   * This step handles the addition of "primary wallets"
+   * which are the only wallets allowed to create new accounts under it
+   */
+  const v2 = async () => {
+    const { wallets, selected } = store.getState().wallets;
+
+    // Check if we have a primary wallet
+    const primaryWallet = Object.keys(wallets).find(key => {
```
Comment (jinchung) [inline src/model/migrations.js:84]: can we use lodash: findKey(wallets, ['primary', true]);

File: src/model/wallet.js [hunk 3]
```diff
@@ -391,12 +391,33 @@ export const createWallet = async (seed = null, color = null, name = null) => {
...
+    // If it's not imported or it's the first one with a seed phrase
+    // it's the primary wallet
+    if (
+      !isImported ||
+      (Object.keys(allWallets) === 0 && type === WalletTypes.mnemonic)
+    ) {
+      primary = true;
+      // Or there's no other primary wallet and this one has a seed phrase
+    } else {
+      const primaryWallet = Object.keys(allWallets).find(key => {
```
Comment (jinchung) [inline src/model/wallet.js:404]: same here, can we use lodash: findKey(allWallets, ['primary', true]);

File: src/model/migrations.js [hunk 4]
```diff
@@ -51,20 +58,76 @@ export default async function runMigrations() {
...
+            primaryWalletKey = key;
+            return true;
+          }
+          return false;
+        });
+      }
+
+      if (primaryWalletKey) {
+        const updatedWallets = { ...wallets };
+        updatedWallets[primaryWalletKey].primary = true;
```
Comment (jinchung) [inline src/model/migrations.js:118]: Given `...wallets` is a shallow copy of the redux wallets object, the object at `updatedWallets[primaryWalletKey]` is still the same object and will complain about immutability when you try to set `.primary = true` on it. May need to do something like `updatedWallets[primaryWalletKey] = { ...updatedWallets[primaryWalletKey], primary: true }`

File: src/model/wallet.js [hunk 5]
```diff
@@ -391,12 +391,33 @@ export const createWallet = async (seed = null, color = null, name = null) => {
       }
     }
 
+    let primary = false;
+    // If it's not imported or it's the first one with a seed phrase
+    // it's the primary wallet
+    if (
+      !isImported ||
+      (Object.keys(allWallets) === 0 && type === WalletTypes.mnemonic)
```
Comment (jinchung) [inline src/model/wallet.js:399]: Object.keys(allWallets) === 0 ?

**Pull request merged**